integration-tests/tests/src/tests/list.ts

```diff
@@ -0,0 +1,2109 @@
+    children: { type: "list", objectType: "PersonObject" },
+    parents: { type: "linkingObjects", objectType: "PersonObject", property: "children" },
+  },
+};
+
+const LinkTypeSchema: Realm.ObjectSchema = {
+  name: "LinkTypesObject",
+  properties: {
+    objectCol: "TestObject",
+    objectCol1: { type: "object", objectType: "TestObject" },
+    arrayCol: "TestObject[]",
+    arrayCol1: { type: "list", objectType: "TestObject" },
+  },
+};
+
+const TestObjectSchema: Realm.ObjectSchema = {
+  name: "TestObject",
+  properties: {
+    doubleCol: "double",
+  },
+};
+
+const PrimitiveArraysSchema: Realm.ObjectSchema = {
+  name: "PrimitiveArrays",
+  properties: {
+    bool: "bool[]",
+    int: "int[]",
+    float: "float[]",
+    double: "double[]",
+    string: "string[]",
+    date: "date[]",
+    data: "data[]",
+    decimal128: "decimal128[]",
+    objectId: "objectId[]",
+    uuid: "uuid[]",
+
+    optBool: "bool?[]",
+    optInt: "int?[]",
+    optFloat: "float?[]",
+    optDouble: "double?[]",
+    optString: "string?[]",
+    optDate: "date?[]",
+    optData: "data?[]",
+    optDecimal128: "decimal128?[]",
+    optObjectId: "objectId?[]",
+    optUuid: "uuid?[]",
+  },
+};
+
+const PersonListSchema: Realm.ObjectSchema = {
+  name: "PersonList",
+  properties: {
+    list: "PersonObject[]",
+  },
+};
+const UuidListSchema: Realm.ObjectSchema = {
+  name: "PrimUuidListsObject",
+  primaryKey: "_id",
+  properties: {
+    _id: "uuid",
+    list: "uuid[]",
+  },
+};
+
+const TargetSchema: Realm.ObjectSchema = {
+  name: "Target",
+  properties: { value: "int" },
+};
+const MidSchema: Realm.ObjectSchema = {
+  name: "Mid",
+  properties: { value: "int", link: "Target" },
+};
+const ListSchema: Realm.ObjectSchema = {
+  name: "List",
+  properties: { list: { type: "list", objectType: "Mid" } },
+};
+
+const NullableBasicTypesSchema: Realm.ObjectSchema = {
+  name: "NullableBasicTypesObject",
+  properties: [
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "boolCol", type: "bool?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "intCol", type: "int?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "floatCol", type: "float?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "doubleCol", type: "double?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "stringCol", type: "string?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "dateCol", type: "date?" },
+    //@ts-expect-error TYPEBUG: should add the field "name" to our interface "ObjectSchemaProperty"
+    { name: "dataCol", type: "data?" },
+  ],
```

Comment:
The API allowing an array to be passed through `properties` has been removed.

```suggestion
  properties: {
    boolCol: "bool?",
    intCol: "int?",
    floatCol: "float?",
    doubleCol: "double?",
    stringCol: "string?",
    dateCol: "date?",
    dataCol: "data?"
  },
```

Reply:
It hasn't been removed from `master` though, right?